### DISCUSSION TITLE
Improvements to Raster Brush Smoothing

### DISCUSSION THREAD

**Discussion by DWrightAnimation** [keiwando/toonsquid#632] [Ideas]
The vector brushes have better smoothing than the raster brushes. For vector brushes, it's like the brush is delayed a bit, making the final result smoother and more accurate to what you're trying to draw, but with raster brushes, it isn't the same, it's like for that, I have to use it very slowly to get the result I want, otherwise it just makes a scribbled mess when trying to draw circles multiple times in the same spot very fast.

**Comment by Cramos821** [comment]
I gotta disagree with you here. Smoothing with vectors feels very laggy to me, while raster smoothing feels very responsive but still smooth. I’m hoping vectors get more responsive smoothing tbh

**Reply by Rohitgit2412** [reply]
But even in Adobe fresco, the vector brushes are laggy. Is this lagginess inherent property of vector brushes as it is common across all the ipad app?

**Reply by DWrightAnimation** [reply]
When I have both smoothing options on for a vector brush, it works like a window stabilizer, which is what I need when trying to make precise lines. However, without the vector smoothing delaying the brush a bit, drawing fast circles with the smoothing set to 100 causes the drawing not to be what I want. If there were more stabilization options, such as a rope stabilizer, a window stabilizer, and a weight stabilizer, there could be more customization for stabilizing your lines.

**Comment by Cat-Butler** [comment]
I wonder if it's at least partly an artefact of rendering speed and Smoothing settings versus specific CPU/GPU combinations. 
On my iPad Mini 5, the default vector brushes seem to be as fast as the pixel brushes, with default settings. 

An increased Smoothing setting can make a huge difference in brush speed (as well as smoothness) for both vector and pixel brushes.
If I bump up the Smoothing on the Fine Oil brush, for example, it slows right dowwwnnn 🐌 (but my shaky drawing gets appropriately smoothed).

Different brushes will require different Smoothing settings, but often around 30-40% makes a significant difference (sometimes too much, but I like really smooth curves so it's often a good starting point). Can be as good as any rope stabiliser I've used, and sometimes better than other window stabilisers.
Above about 40% (more or less), it becomes a real balancing act between smoothing and retaining the desired drawn line shape (over-smoothing over-simplifies the drawn line shape, which is often not what you want).

I'm curious, are you talking about concentric circles? Are you layering transparencies? I'm now wondering what is the application of "draw circles multiple times in the same spot very fast".

**Comment by DWrightAnimation** [comment]
Looks like in the next update, 2.0.6, there will be additional brush smoothing settings that will make the raster brushes work like the vector brushes in terms of smoothing, while also retaining the option for the way it currently is. As for the vector brushes, it seems like whether you have the setting that makes the raster brushes smooth like the vector brushes on or off in a vector brush, it seems to be on regardless.
